src/strategies/StrategyFactory.h

```diff
@@ -0,0 +1,34 @@
+/**
+ * @file StrategyFactory.h
+ * @brief Factory class for choosing the appropriate functions based on the Simulation.
+ * @date 2024-11-11
+ *
+ * @copyright Copyright (c) 2024
+ *
+ */
+
+#pragma once
+#include "objects/ParticleContainer.h"
+#include "utils/Arguments.h"
+#include <functional>
+#include <tuple>
+
+/// @brief Factory class for choosing the appropriate functions based on the Simulation.
+class StrategyFactory {
+  public:
+    /// @brief Typedef for velocity-calculating functions.
+    using VFunc = void (*)(ParticleContainer &, double);
+    /// @brief Typedef for position-calculating functions.
+    using XFunc = void (*)(ParticleContainer &, double);
+    /// @brief Typedef for force-calculating functions.
+    using FFunc = void (*)(ParticleContainer &, double, double);
+
+    /**
+     * @brief Return a 3-tuple of the physics functions corresponding to the chosen simulation.
+     *
+     * @param type The type of the simulation to be performed.
+     * @param modifier A modifier for which combination of functions get returned (i.e. optimized, non-optimized).
+     * @return A 3-tuple of the physics functions corresponding to the chosen simulation.
+     */
+    static std::tuple<VFunc, XFunc, FFunc> getSimulationFunctions(SimulationType type, int modifier);
```

Comment:
From what I understood from this comment, `VFunc` and `XFunc` should somehow be structurally linked to eachother to avoid explicitly choosing one without the other in the context of a simulation. I'm still not fully convinced I _entirely_ fixed it, but now, the position and force calculations have been moved to a new `struct TimeIntegrationFuncs` with a constructor which takes a simulation type as a parameter and sets the member variables (representing the two functions) accordingly - at least now, they're somehow connected and no longer as independent as `FFunc`, for example.

Still, this shouldn't be too big of an issue either way, since at the end of the day, it's up to us, the developers, to ensure that the correct functions are chosen. I guess this helps to represent the mathematical correctness more than anything.